kasa/discover.py

```diff
@@ -23,16 +23,16 @@
 )
 from kasa.deviceconfig import ConnectionType, DeviceConfig, EncryptType
 from kasa.exceptions import TimeoutException, UnsupportedDeviceException
+from kasa.iot.device import Device, SmartDeviceException
```

Comment:
The exception should be imported from `kasa.exceptions`. I'm wondering if we should consider renaming it also already in this PR?

Reply:
Done the import change.

I agree with changing it to `DeviceException` but I think we should do this in a PR immediately after this one because it changes nearly every file so it'll be easier to review separately and not confuse this one.

Reply:
Resolving as we've agreed to do in a separate PR